### PR TITLE
AddrSplosion III - Revenge of the granular TTL

### PR DESCRIPTION
**Pull request by jbenet**
This PR adds a granular TTL AddrManager. This way,
every time we add an address, we specify the intended
duration of the addr.

This PR:
- associates addrs with a TTL
- rate limits dials (10 per peer)
- adds an `ipfs swarm addrs` command for monitoring known addrs
- bumps the protocol version
- disconnects from clients whose protocol version mismatches

### CODE REVIEW

File: p2p/protocol/identify/id.go [hunk 1]
```diff
@@ -176,12 +181,22 @@ func (ids *IDService) consumeMessage(mes *pb.Identify, c inet.Conn) {
...
-	ids.Host.Peerstore().SetAddresses(p, lmaddrs)
+	ids.Host.Peerstore().SetAddrs(p, lmaddrs, peer.ConnectedAddrTTL)
 	log.Debugf("%s received listen addrs for %s: %s", c.LocalPeer(), c.RemotePeer(), lmaddrs)
 
 	// get protocol versions
 	pv := mes.GetProtocolVersion()
 	av := mes.GetAgentVersion()
+
+	// version check. if we shouldn't talk, bail.
+	// TODO: at this point, we've already exchanged information.
```
Comment (whyrusleeping) [inline p2p/protocol/identify/id.go]: how far off of a TODO is this?

**Comment by whyrusleeping** [comment]
LGTM!

File: p2p/protocol/identify/id.go (hunk 1 above)
Comment (jbenet) [inline p2p/protocol/identify/id.go]: post alpha

**Comment by jbenet** [comment]
# :warning: protocol version bump

after this merges you wont be able to nodes with prior versions

**Pull request merged**